base/linalg/blas.jl

```diff
@@ -339,6 +369,15 @@ nrm2(x::Array) = nrm2(length(x), pointer(x), 1)
     asum(n, X, incx)
 
 Sum of the absolute values of the first `n` elements of array `X` with stride `incx`.
+
+# Example:
+```jldoctest
+julia> Base.BLAS.asum(10, im*ones(10), 2)
+5.0
```

Comment:
If you make `a` longer though, you'll get results that read from past the 6th element